src/CircleEIP1193Provider.ts

```diff
@@ -36,13 +48,32 @@ class CircleEIP1193Provider {
     }
   }
 
+  async createWalletSet(publicKey: string): Promise<string | undefined> {
+    try {
+      const encryptedEntitySecret = encryptEntitySecret(publicKey);
+      const data = {entitySecretCiphertext: encryptedEntitySecret, idempotencyKey: '85caf545-770c-429a-8eaa-72d8f01974c3', name: 'Wallet Set A'};
```

Comment:
I would pass the `encryptedEntitySecret` directly into `data` and you need to use a function to generate the UUID for `idempotencyKey`, else the API will treat every request as the same request. 
```suggestion
  async createWalletSet(): Promise<string | undefined> {
    try {
      const data = {entitySecretCiphertext: encryptEntitySecret(this.publicKey), idempotencyKey: genUUID(), name: 'Wallet Set A'};
```